x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/SearchableSnaposhotActionTests.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ilm;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.admin.cluster.snapshots.status.SnapshotsStatusAction;
+import org.elasticsearch.action.admin.cluster.snapshots.status.SnapshotsStatusRequest;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.common.TriConsumer;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.test.client.NoOpClient;
+import org.elasticsearch.xpack.core.ilm.OnAsyncWaitBranchingStep.BranchingStepListener;
+import org.elasticsearch.xpack.core.ilm.Step.StepKey;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.xpack.core.ilm.SearchableSnapshotAction.NAME;
+import static org.elasticsearch.xpack.core.ilm.SearchableSnapshotAction.getCheckSnapshotStatusAsyncAction;
+import static org.hamcrest.Matchers.is;
+
+public class SearchableSnaposhotActionTests extends AbstractActionTestCase<SearchableSnapshotAction> {
```

Comment:
Super minor, but there's a typo here (and in the name of the file):
`SearchableSnaposhotActionTests` vs
`SearchableSnapshotActionTests`

Reply:
Ah, it was hard to spot the typo, great eyes, thanks @dakrone ! 

Regarding the action name, I renamed it to "searchable_snapshot" (lowercase) on purpose when we renamed "snapshot-repository" to "snapshot_repository" as I thought it was confusing to have a configuration that mixes scores and underscores eg.:
```
  "searchable-snapshot" : { 
    "snapshot_repository" : "snapshotRepositoryName"
  }
```
I using underscores everywhere employs less cognitive load: 
```
  "searchable_snapshot" : { 
    "snapshot_repository" : "snapshotRepositoryName"
  }
```

Reply:
The other "composed" action we have is `set_priority` so it would be inconsistent to use scores. If you don't have a strong opinion on this, I think I'd prefer we use underscores everywhere.

Reply:
I read the other one as the step name rather than the action name, I agree that the action name should use underscores (like our other ones), sorry about that!